federationapi/routing/send.go

```diff
@@ -367,53 +365,31 @@ func (t *txnReq) processEvent(ctx context.Context, e gomatrixserverlib.Event) er
 		return roomNotFoundError{e.RoomID()}
 	}
 
-	if len(stateResp.MissingAuthEventIDs) > 0 {
-		logger.Infof("Event refers to %d unknown auth_events", len(stateResp.MissingAuthEventIDs))
-
-		servers := []gomatrixserverlib.ServerName{t.Origin}
+	// We will need to know this when fetching missing auth or prev events.
+	getServers := func() {
+		t.servers = []gomatrixserverlib.ServerName{}
```

Comment:
Feels like a peculiar form, why have this as a nested function which side effects by setting `t.servers`? Why not just have `t.getServers() []ServerName`?

Reply:
I've refactored it to be a bit less strange!